lib/orderbook/OrderBook.ts

```diff
@@ -96,19 +98,26 @@ class OrderBook extends EventEmitter {
     return this.addOwnOrder(order);
   }
 
-  public addMarketOrder = (order: orders.MarketOrder) => {
+  public addMarketOrder = (order: orders.OwnMarketOrder): matchingEngine.MatchingResult => {
     const price = order.quantity > 0 ? Number.MAX_VALUE : 0;
     return this.addOwnOrder({ ...order, price }, true);
   }
 
-  public removeOwnOrder = (orderId: string, pairId: string): boolean => {
+  public removeOwnOrder = (pairId: string, orderId: string): boolean => {
     const matchingEngine = this.matchingEngines[pairId];
     if (!matchingEngine) {
       throw errors.INVALID_PAIR_ID(pairId);
     }
 
-    if (matchingEngine.removeOwnOrder(orderId)) {
-      return this.removeOrder(this.ownOrders, orderId, pairId);
+    let id = orderId;
+
+    if (this.localIds[orderId]) {
```

Comment:
Rather than doing this local id lookup every time we remove an order, I think it's better to create a wrapper method called `removeOwnOrderByLocalId` which takes `pairId` and `localId`, performs this lookup, and then calls and returns the result of `removeOwnOrder`. If it doesn't find a matching value for `localId` it returns false.